public/mobile-app/src/routes/settings/+page.svelte

```diff
@@ -17,8 +16,13 @@
     isChecked = localStorage.getItem('notifications_enabled') === 'true'
   })
 
-  const navigateToPreviousPage = async () => {
-    window.history.back()
+  const navigateToPreviousPageOrHomepage = async () => {
+    if (localStorage.getItem('user_first_login')) {
+      localStorage.removeItem('user_first_login')
```

Comment:
I believe this is the only place where we remove this item, but I'm afraid this might bite us in the future, I'm not feeling very confident about that...